### DISCUSSION TITLE
Odd Behavior - A Daily 7am Solar Dropout

### DISCUSSION THREAD

**Discussion by jasonacox** [jasonacox/Powerwall-Dashboard#35] [Q&A]
For some reason my system stops solar production at 7:00am PT every day for about 5 minutes.   At first, I figured this was a way for it to balance the strings (and still may be) but it is odd that it doesn't seem to relate to the amount of solar production at the time (often above 1kW).  Has anyone else seen this?

<img width="565" alt="Screen Shot 2022-06-11 at 8 56 38 AM" src="https://user-images.githubusercontent.com/836718/173201746-6168158d-e8cf-4d96-ad0d-06807cbc528b.png">

**Comment by cwagz** [comment]
Does your powerwall charge level graph also drop off at the same time? I was seeing this for a few days several weeks ago but I believe it was due to a firmware update and reboot. Everything has been solid for awhile now. 

Seems to have stopped for me after 22.9.2 was installed.

**Reply by jasonacox** [reply]
Interesting!  Thanks. 

The solar, inverter and strings graphs are the only ones dropping.  I'm on 22.9.2 a54352e6 so it does seems like this might be something different.  It is odd that it is consistently 7am.  Also, it looks like the voltage doesn't drop completely for all the strings. I wonder if this is some MTTP calibration event.  Another theory I have, the installers hooked A, C and D strings (no connect for B) and the system could be doing a retry to power up B, but even if that were the case it is odd that it is happening at 7am every day, even on cloudy days.  I'm going to try to spend a few days watching it closely during 7am to see if I spot anything new.  Of course, it isn't lost on me that this is a < 5 minute event, but it just still bugs me. :)

<img width="1250" alt="image" src="https://user-images.githubusercontent.com/836718/173751227-8c86e776-ef32-44c6-9c1f-22a27005f0fe.png">

**Reply by biteableniles** [reply]
Seeing same behavior on 22.9.2 a54352e6. Two powerwalls wired on A, B, and C strings. All string and inverter output drops to zero then recovers fully. Have only been collecting data for three days but has happened each day.
![Powerwall 7am](https://user-images.githubusercontent.com/107505688/173836637-f8366234-1bab-4a52-a3cb-32c964abaa8d.png)

**Comment by cwagz** [comment]
Actually my system does dip right at 7am. I thought this was a shadow!  It’s not dipping to zero but it’s there. 

![3B2ED253-DF74-4A1C-807C-8187D6D11E3F](https://user-images.githubusercontent.com/13752647/173863156-959bae4b-76b7-4285-ae8f-e99b2742eed2.jpeg)

**Reply by jasonacox** [reply]
I watched the vitals (http://localhost:8675/vitals) this morning shortly after 7am and this is what I noticed an Alert when solar dropped to zero at 7:01:

    PVAC_a014_PVS_disabled_relay

I also noticed the following state changes:

| **Vital Item** | **7:00** | **7:01** | **7:03** | **7:04** |
| :--- | :---: | :---: | :---: | :---: | 
|  | | | | |
| `PVAC_InvState` | INV_Grid_Connected | INV_Disconnected | INV_Disconnected | INV_Grid_Connected |
| `PVAC_State` | PVAC_Active| PVAC_Standby | PVAC_Standby | PVAC_Active|
|  | | | |
| `PVS_EnableOutput` | true |false | false | true |
| `PVS_SelfTestState` | PVS_SelfTestOff | PVS_SelfTesting | PVS_SelfTesting | PVS_SelfTestOff |
| `PVS_State` | PVS_Active |PVS_SelfTestMci | PVS_SelfTestIsolation | PVS_Active |

I don't know why it suddenly activates a `PVS_SelfTestMci` - which we normally see as the system comes online at first light.  It could be that this is 7am timer to bump the MCI which is the Tesla PV Rapid Shutdown devices on the strings.  I guess this could be the result of instances where some string MCIs don't start up during normal first light and stay offline (reduced full day capacity).  It still feels like a bug. ;)

**Comment by oralallen82** [comment]
> For some reason my system stops solar production at 7:00am PT every day for about 5 minutes. At first, I figured this was a way for it to balance the strings (and still may be) but it is odd that it doesn't seem to relate to the amount of solar production at the time (often above 1kW). Has anyone else seen this?
> 
> <img alt="Screen Shot 2022-06-11 at 8 56 38 AM" width="565" src="https://user-images.githubusercontent.com/836718/173201746-6168158d-e8cf-4d96-ad0d-06807cbc528b.png">

This is due to the self test that the system performs if you log into the Powerwall +, right at 7am you'll see self test. It takes roughly between 3-5 minutes. I like this method as before it was 24 hours +10 minutes from the last system reset. Eventually it'll end up being during peak performance. While it's less than 10 minutes it wasn't desired.

**Reply by jasonacox** [reply]
Oh wow!  Thanks @oralallen82 - I had never noticed the 24+10m blips.  But agree, if there is a plan cycle, put it at the low production times.   I thought this was happening at first light too, but perhaps the intent here is to do a test when there is more solar power to fully test?

**Reply by oralallen82** [reply]
> Oh wow! Thanks @oralallen82 - I had never noticed the 24+10m blips. But agree, if there is a plan cycle, put it at the low production times. I thought this was happening at first light too, but perhaps the intent here is to do a test when there is more solar power to fully test?

Looking at the timing and some of the tests, it does it does relay on some production.

**Comment by jasonacox** [comment]
With the recent Powerwall Firmware update to 22.18.3 I noticed that the "test" moved from 7am to 11:55am (a higher solar production time so not great), but it has since moved back to 7am after a few days:

<img width="616" alt="image" src="https://user-images.githubusercontent.com/836718/177007929-b27767ea-0827-4511-8c5f-0dff6fc3cdb2.png">

The test does seem to be a bit faster.  I was seeing 5m outages and it is 3m,  to the point that zoomed out on most graphs you won't even see it.  To be fair, small data sample set (2 days).  Others are reporting no test showing up now.

<img width="610" alt="image" src="https://user-images.githubusercontent.com/836718/177010376-c3300e61-3659-4eb1-b685-75d6a95e2e93.png">

**Comment by oralallen82** [comment]
Odd. I don't have that issue. I do not see that blip on mine after that update.

**Reply by jasonacox** [reply]
@oralallen82 out of curiosity, are your strings contiguous? In other words, did they install all active strings together without gaps?  The installers put my standalone string in A and then a parallel string in C & D, resulting in a gap (B) which does throw an alert every morning or during the test.  I wonder if that triggers the system to try a test later in the day.  As a note, in the design documents, it said they were supposed to put the parallel strings as A & B and the stand alone as C (which seem more elegant in my OCD opinion 😁 ). 

http://localhost:8675/strings

```json
{
    "A": {
        "Connected": true,
        "Current": 6.8,
        "Power": 1547.0,
        "State": "PV_Active",
        "Voltage": 228.0
    },
    "B": {
        "Connected": false,
        "Current": 0.12,
        "Power": 0.0,
        "State": "PV_Active",
        "Voltage": -1.9000000000000004
    },
    "C": {
        "Connected": true,
        "Current": 8.6,
        "Power": 2228.0,
        "State": "PV_Active",
        "Voltage": 261.1
    },
    "D": {
        "Connected": true,
        "Current": 8.58,
        "Power": 2226.0,
        "State": "PV_Active_Parallel",
        "Voltage": 261.1
    }
}
```

**Reply by oralallen82** [reply]
I do have contiguous strings, West, South and East. A, B, C.  That's odd they hooked them up like that.
